File: ChatGPT/ChatGPT/javascript/tcpClient.js
const net = require('net');

const SERVER_IP = '127.0.0.1';
const SERVER_PORT = 12345;

const client = new net.Socket();

client.connect(SERVER_PORT, SERVER_IP, () => {
    console.log('서버에 연결되었습니다.');
    client.write('Hello, server!');
});

client.on('data', (data) => {
    console.log('서버로부터 받은 응답:', data.toString());
    client.destroy();
});

client.on('close', () => {
    console.log('서버 연결이 종료되었습니다.');
});

File: ChatGPT/ChatGPT/java/tcpClient.java
import java.io.*;
import java.net.*;

public class TCPClient {
    public static void main(String[] args) {
        final String SERVER_IP = "127.0.0.1";
        final int SERVER_PORT = 12345;

        try {
            // TCP 서버에 연결
            Socket socket = new Socket(SERVER_IP, SERVER_PORT);
            System.out.println("서버 연결 성공");

            // 서버로 메시지 전송
            String message = "Hello, server!";
            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);
            out.println(message);
            System.out.println("메시지 전송 성공");

            // 소켓과 스트림 닫기
            out.close();
            socket.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
